axon_sql.go

```diff
@@ -25,13 +24,10 @@ func prepareQueryArgs(changesetCols []*ChangesetColumn) ([]string, []string, map
 	for _, c := range changesetCols {
 		t := reflect.TypeOf(c.Value)
 		if t != nil && t.Kind() == reflect.Map {
-			// Found a hashmap, this is a JSON/B field. Convert manually to string to
-			// avoid package sql error: "unsupported type map[string]interface {}".
-			b, err := json.Marshal(c.Value)
-			if err != nil {
-				return cols, colArgs, values, fmt.Errorf("unable to marshal JSON field %s: %w", c.Column, err)
-			}
-			c.Value = string(b)
+			// Found a hashmap, this is a JSON/B field. This type is not supported
```

Comment:
If we are already handling map[string]interface{} (which is when the database field has some sort of key/value data structure - json or jsonb afaik) in the listener, do we need this check here? Maybe we can move the comments to the listener?

Reply:
I think this is the correct place to check and return an error. It prevents anyone writing their own custom listener and breaking expected behavior. If it were not for breaking verify, I would not worry about handling it.